src/cryptonote_core/service_node_list.cpp

```diff
@@ -2168,5 +2168,18 @@ namespace service_nodes
     cmd = stream.str();
     return true;
   }
+
+  bool service_node_info::can_transition_to_state(new_state proposed_state) const
+  {
+    new_state last_state = last_state_transition();
```

Comment:
I'm not sure you need to care about the last state tx, but only the current state.  I think the only state change txes that are invalid are:

- (decomm or ip penalty) if currently decommissioned
- recommission if not currently decommissioned

It seems like it would be easier to just fail on those specific cases (and not need the `last_state_transition()` or `null_state` addition).

Reply:
Yes, that does need less code. Updated